churaverse-plugins-server/src/gamePlugin/domain/coreGamePlugin.ts

```diff
@@ -96,4 +99,31 @@ export abstract class CoreGamePlugin extends BaseGamePlugin implements IGameInfo
     this._participantIds = []
     this.gamePluginStore.games.delete(this.gameId)
   }
+
+  private readonly onPlayerLeave = (ev: PlayerLeaveEvent): void => {
+    if (ev.id === undefined) return
+
+    if (this.removeParticipant(ev.id)) {
+      this.handlePlayerLeave(ev.id)
+    }
+  }
+
+  private removeParticipant(playerId: string): boolean {
+    const idx = this._participantIds.indexOf(playerId)
+    if (idx === -1) return false
+    this._participantIds.splice(idx, 1)
+    this.sendParticipantUpdate()
+    return true
+  }
```

Comment:
removeParticipant関数の役割は`_participantIds`配列の更新を行うことだと思います。
なので、removeParticipant関数でsendParticipantUpdateを呼び出すのではなく、onPlayerLeave関数内で呼び出すのが適切かなと思います。

```suggestion
  private readonly onPlayerLeave = (ev: PlayerLeaveEvent): void => {
    if (ev.id === undefined) return

    if (this.removeParticipant(ev.id)) {
      this.sendParticipantUpdate()
      this.handlePlayerLeave(ev.id)
    }
  }

  private removeParticipant(playerId: string): boolean {
    const idx = this._participantIds.indexOf(playerId)
    if (idx === -1) return false
    this._participantIds.splice(idx, 1)
    return true
  }
```